### ISSUE TITLE
[Feature request] Equivalent of softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by rohitrango** [pytorch/pytorch#8199]
I don't think there an equivalent of `tf.nn.softmax_cross_entropy_with_logits` in PyTorch. The closest function I found is `BCEWithLogitsLoss` but it combines a `sigmoid` followed by the BCE loss instead of a `softmax`. 
Can we have an implementation of a `softmax_cross_entropy_with_logits` (instead of combining softmax and cross entropy loss for numerical stability)? 

Thanks!

**Comment by soumith** [comment]
you are looking for `nn.CrossEntropyLoss`. It is nn.LogSoftmax + nn.NLLLoss

**Comment by henribru** [comment]
But those aren't equivalent. CrossEntropyLoss expects class indices, while softmax_cross_entropy_with_logits can work with probability distributions.
